src/app/(sidebar)/smart-contracts/contract-explorer/page.tsx

```diff
@@ -169,6 +169,23 @@ export default function ContractExplorer() {
     ) : null;
   };
 
+  const handleLoadContract = async () => {
+    try {
+      // throwOnError will throw an error instead of setting the error state.
+      // This will prevent other queries from running when fetching a contract
+      // type fails.
+      await fetchContractType({ throwOnError: true });
+      await fetchContractInfo({ throwOnError: true });
+      await fetchWasmContractClient();
```

Comment:
With this update, if the `fetchContractType` fails, `fetchContractInfo` won't be called. Previously, we made both calls, which wasn't optimal.